pkg/microservice/aslan/core/label/handler/labels.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2022 The KodeRover Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handler
+
+import (
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/koderover/zadig/pkg/microservice/aslan/core/label/service"
+	internalhandler "github.com/koderover/zadig/pkg/shared/handler"
+	e "github.com/koderover/zadig/pkg/tool/errors"
+)
+
+func ListLabels(c *gin.Context) {
+	ctx := internalhandler.NewContext(c)
+	defer func() { internalhandler.JSONResponse(c, ctx) }()
+	listLabelsArgs := new(service.ListLabelsArgs)
+	if err := c.ShouldBindJSON(&listLabelsArgs); err != nil {
+		ctx.Err = err
+		return
+	}
+	ctx.Resp, ctx.Err = service.ListLabels(listLabelsArgs)
+}
+
+func createLabelValidate(lbs *service.CreateLabelsArgs) error {
+	for _, v := range lbs.Labels {
+		if v.Key == "" || v.Value == "" {
+			return e.ErrInvalidParam.AddDesc("invalid label args")
+		}
+	}
+	return nil
+}
+
+func CreateLabels(c *gin.Context) {
+	ctx := internalhandler.NewContext(c)
+	defer func() { internalhandler.JSONResponse(c, ctx) }()
+
+	labels := new(service.CreateLabelsArgs)
+	if err := c.ShouldBindJSON(&labels); err != nil {
+		ctx.Err = err
+		return
+	}
+
+	if err := createLabelValidate(labels); err != nil {
+		ctx.Err = err
+		return
+	}
+
+	ctx.Resp, ctx.Err = service.CreateLabels(labels, ctx.UserName)
+}
+
+//DeleteLabels  can only bulk delete labels which not bind reousrces
```

Comment:
`DeleteLabels  can only bulk delete labels which not bind reousrces` -> `DeleteLabels can only bulk delete labels which do not bind resources`

Reply:
fixed